dll/win32/browseui/shellbars/CISFBand.cpp

```diff
@@ -13,100 +14,192 @@
 #include <wingdi.h>
 #include <uxtheme.h>
 
+#undef SubclassWindow // Don't use SubclassWindow macro
+
 /*
 TODO:
     ** drag and drop support
     ** tooltips
-    ** handle change notifications
     ** Fix position of the items context menu
     ** Implement responding to theme change
 */
 
+#define TIMERID_DELAYED_REFRESH 0xBEEFCAFE
+#define TIMER_REFRESH_DELAY 500
+
 //*****************************************************************************************
 // *** CISFBand ***
 
-CISFBand::CISFBand() :
-    m_BandID(0),
-    m_pidl(NULL),
-    m_textFlag(true),
-    m_iconFlag(true),
-    m_QLaunch(false)
+CISFBand::CISFBand()
+    : m_BandID(0)
+    , m_uChangeNotify(0)
+    , m_bShowText(TRUE)
+    , m_bSmallIcon(TRUE)
+    , m_QLaunch(FALSE)
 {
 }
 
 CISFBand::~CISFBand()
 {
-    CloseDW(0);
 }
 
-// Toolbar
-/*++
-* @name CreateSimpleToolbar
-*
-* Creates a toolbar and fills it up with buttons for enumerated objects.
-*
-* @param hWndParent
-*        Handle to the parent window, which receives the appropriate messages from child toolbar.
-*
-* @return The error code.
-*
-*--*/
 HRESULT CISFBand::CreateSimpleToolbar(HWND hWndParent)
 {
-    // Declare and initialize local constants.
-    const DWORD buttonStyles = BTNS_AUTOSIZE;
-
     // Create the toolbar.
-    m_hWnd = CreateWindowEx(0, TOOLBARCLASSNAME, NULL,
-        WS_CHILD | TBSTYLE_FLAT | TBSTYLE_LIST | CCS_NORESIZE | CCS_NODIVIDER, CW_USEDEFAULT, CW_USEDEFAULT, 0, 0,
-        hWndParent, NULL, 0, NULL);
-    if (m_hWnd == NULL)
+    DWORD style = WS_CHILD | TBSTYLE_FLAT | TBSTYLE_LIST | CCS_NORESIZE | CCS_NODIVIDER;
+    HWND hwndToolbar = ::CreateWindowExW(0, TOOLBARCLASSNAMEW, NULL, style, 0, 0, 0, 0,
```

Comment:
So, to follow up with my previous question about m_hwnd, is the m_hwnd equal to hwndtoolbar or not? How do they relate? If they don't, which window does m_hwnd correspond to?

Reply:
> is the m_hwnd equal to hwndtoolbar or not?

Yes, it is. `SubclassWindow` sets `m_hWnd` and does hook the window procedure.